vocabularies/DirectEdit.json

```diff
@@ -0,0 +1,71 @@
+{
+    "$Version": "4.01",
+    "$Reference": {
+        "https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.json": {
+            "$Include": [
+                {
+                    "$Namespace": "Org.OData.Core.V1",
+                    "$Alias": "Core"
+                }
+            ]
+        },
+        "https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Validation.V1.json": {
+            "$Include": [
+                {
+                    "$Namespace": "Org.OData.Validation.V1",
+                    "$Alias": "Validation"
+                }
+            ]
+        },
+        "https://sap.github.io/odata-vocabularies/vocabularies/Common.json": {
+            "$Include": [
+                {
+                    "$Namespace": "com.sap.vocabularies.Common.v1",
+                    "$Alias": "Common"
+                }
+            ]
+        }
+    },
+    "com.sap.vocabularies.DirectEdit.v1": {
+        "$Alias": "DirectEdit",
+        "@Core.Description": "Terms for Direct-Edit User Interfaces",
+        "@Core.Description#Published": "2020-07-10 © Copyright 2020 SAP SE. All rights reserved",
+        "@Core.Links": [
+            {
+                "rel": "alternate",
+                "href": "https://sap.github.io/odata-vocabularies/vocabularies/DirectEdit.xml"
+            },
+            {
+                "rel": "latest-version",
+                "href": "https://sap.github.io/odata-vocabularies/vocabularies/DirectEdit.json"
+            },
+            {
+                "rel": "describedby",
+                "href": "https://github.com/sap/odata-vocabularies/blob/master/vocabularies/DirectEdit.md"
+            }
+        ],
+        "SideEffects": {
+            "$Kind": "Term",
+            "$Type": "DirectEdit.SideEffectsType",
+            "$AppliesTo": [
+                "EntitySet"
+            ],
+            "@Common.Experimental": true,
+            "@Core.Description": "Determine side effects of client-side data modification"
+        },
+        "SideEffectsType": {
+            "$Kind": "ComplexType",
+            "@Common.Experimental": true,
+            "@Core.LongDescription": "After a change to a property whose path is contained in `Triggers`, the client should pass the entity with the changed property to the `CalculationFunction` and receive the entity with all changes that happen as side effects of the given change.",
+            "Triggers": {
+                "$Collection": true,
+                "@Core.Description": "List of paths to the properties whose changes should trigger the side effects calculation.",
+                "@Validation.MinItems": 1
+            },
+            "CalculationFunction": {
```

Comment:
In a more complex UI, the function might not only return changes to the entity itself but also for dependent entities. E.g. when called on a Sales Order also changes to items might be returned. 
The function call would then need to be done with $expand. How would that work out?